### PR TITLE
Feature/iesp 171

### PR DESCRIPTION
**Pull request by AlessandroColucci**
Refactoring of api for probing data to be shown on detail page based on the modifications made on operations. Inserted the presentational logic from operations. Cleaning of code.

### CODE REVIEW

File: src/main/java/it/pagopa/interop/probing/probingapi/rest/ProducerController.java [hunk 1]
```diff
@@ -6,17 +6,17 @@
...
 import it.pagopa.interop.probing.probingapi.dtos.SearchProducerNameResponse;
-import it.pagopa.interop.probing.probingapi.service.EserviceService;
+import it.pagopa.interop.probing.probingapi.service.ProducerService;
 
 @RestController
 public class ProducerController implements ProducersApi {
 
   @Autowired
-  EserviceService eserviceService;
+  ProducerService producerService;
```
Comment (nttdata-rtorsoli) [inline src/main/java/it/pagopa/interop/probing/probingapi/rest/ProducerController.java]: use private

File: src/main/java/it/pagopa/interop/probing/probingapi/mapping/mapper/AbstractMapper.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import it.pagopa.interop.probing.probingapi.dtos.EserviceStateFE;
+import it.pagopa.interop.probing.probingapi.dtos.ProbingDataEserviceResponse;
+import it.pagopa.interop.probing.probingapi.dtos.SearchEserviceContent;
+import it.pagopa.interop.probing.probingapi.mapping.dto.impl.ProbingDataEserviceBEResponse;
+import it.pagopa.interop.probing.probingapi.mapping.dto.impl.SearchEserviceBEContent;
+import it.pagopa.interop.probing.probingapi.util.EnumUtilities;
+
+@Mapper(componentModel = "spring")
+public abstract class AbstractMapper {
```
Comment (nttdata-rtorsoli) [inline src/main/java/it/pagopa/interop/probing/probingapi/mapping/mapper/AbstractMapper.java:14]: Now with the introduction of the interfaces on Dto, Request, Response and Entity we can start to create a generic  converter function  using Function<A,B> and rid of the heavy Mapstruct library

**Review by nttdata-rtorsoli — DISMISSED** [review]
See comments

File: src/main/java/it/pagopa/interop/probing/probingapi/rest/ProducerController.java (hunk 1 above)
Comment (AlessandroColucci) [reply src/main/java/it/pagopa/interop/probing/probingapi/rest/ProducerController.java]: ok

File: src/main/java/it/pagopa/interop/probing/probingapi/mapping/mapper/AbstractMapper.java (hunk 2 above)
Comment (AlessandroColucci) [reply src/main/java/it/pagopa/interop/probing/probingapi/mapping/mapper/AbstractMapper.java:14]: ok.I will discuss it with the team.

File: src/main/java/it/pagopa/interop/probing/probingapi/mapping/dto/impl/ProbingDataEserviceBEResponse.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import it.pagopa.interop.probing.probingapi.dtos.EserviceStateBE;
+import it.pagopa.interop.probing.probingapi.mapping.dto.Response;
+import it.pagopa.interop.probing.probingapi.util.EserviceStatus;
+import lombok.Builder;
+import lombok.Data;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
```
Comment (nttdata-rtorsoli) [inline src/main/java/it/pagopa/interop/probing/probingapi/mapping/dto/impl/ProbingDataEserviceBEResponse.java]: getter and setter already included in data annotation

File: src/main/java/it/pagopa/interop/probing/probingapi/mapping/dto/impl/SearchEserviceBEContent.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import it.pagopa.interop.probing.probingapi.mapping.dto.Response;
+import it.pagopa.interop.probing.probingapi.util.EserviceStatus;
+import lombok.Builder;
+import lombok.Data;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@Data
```
Comment (nttdata-rtorsoli) [inline src/main/java/it/pagopa/interop/probing/probingapi/mapping/dto/impl/SearchEserviceBEContent.java:13]: same as above

File: src/main/java/it/pagopa/interop/probing/probingapi/mapping/dto/impl/SearchEserviceBEResponse.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import it.pagopa.interop.probing.probingapi.mapping.dto.Response;
+import lombok.Builder;
+import lombok.Data;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@Data
+@Builder
```
Comment (nttdata-rtorsoli) [inline src/main/java/it/pagopa/interop/probing/probingapi/mapping/dto/impl/SearchEserviceBEResponse.java:12]: same as above

**Review by nttdata-rtorsoli — DISMISSED** [review]

**Review by galales — APPROVED** [review]

File: pom.xml [hunk 6]
```diff
@@ -100,6 +100,32 @@
...
+				<artifactId>maven-compiler-plugin</artifactId>
+				<version>3.8.1</version>
+				<configuration>
+					<source>17</source>
+					<target>17</target>
+					<annotationProcessorPaths>
+						<path>
+							<groupId>org.projectlombok</groupId>
+							<artifactId>lombok</artifactId>
+							<version>1.18.24</version>
```
Comment (nttdata-rtorsoli) [inline pom.xml:114]: use properties

File: pom.xml [hunk 7]
```diff
@@ -100,6 +100,32 @@
...
+						</path>
+						<path>
+							<groupId>org.mapstruct</groupId>
+							<artifactId>mapstruct-processor</artifactId>
+							<version>1.5.1.Final</version>
```
Comment (nttdata-rtorsoli) [inline pom.xml:119]: see above

File: pom.xml [hunk 8]
```diff
@@ -100,6 +100,32 @@
...
+						</path>
+						<path>
+							<groupId>org.projectlombok</groupId>
+							<artifactId>lombok-mapstruct-binding</artifactId>
+							<version>0.2.0</version>
```
Comment (nttdata-rtorsoli) [inline pom.xml:124]: see above

**Review by nttdata-rtorsoli — APPROVED** [review]

**Pull request merged**